File: Example/API/src/realestate/model/RealEstate.java
package realestate.model;

import java.io.Serializable;
import javax.persistence.*;


/**
 * The persistent class for the RealEstate database table.
 * 
 */
@Entity
@NamedQuery(name="RealEstate.findAll", query="SELECT r FROM RealEstate r")
public class RealEstate implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy=GenerationType.IDENTITY)
	@Column(name="id_real_esate")
	private int idRealEsate;

	@Column(name="garden_squaremeter")
	private double gardenSquaremeter;

	private String location;

	@Column(name="num_of_parkinglots")
	private int numOfParkinglots;

	private int rooms;

	private double squaremeter;

	private String type;

	public RealEstate() {
	}

	public int getIdRealEsate() {
		return this.idRealEsate;
	}

	public void setIdRealEsate(int idRealEsate) {
		this.idRealEsate = idRealEsate;
	}

	public double getGardenSquaremeter() {
		return this.gardenSquaremeter;
	}

	public void setGardenSquaremeter(double gardenSquaremeter) {
		this.gardenSquaremeter = gardenSquaremeter;
	}

	public String getLocation() {
		return this.location;
	}

	public void setLocation(String location) {
		this.location = location;
	}

	public int getNumOfParkinglots() {
		return this.numOfParkinglots;
	}

	public void setNumOfParkinglots(int numOfParkinglots) {
		this.numOfParkinglots = numOfParkinglots;
	}

	public int getRooms() {
		return this.rooms;
	}

	public void setRooms(int rooms) {
		this.rooms = rooms;
	}

	public double getSquaremeter() {
		return this.squaremeter;
	}

	public void setSquaremeter(double squaremeter) {
		this.squaremeter = squaremeter;
	}

	public String getType() {
		return this.type;
	}

	public void setType(String type) {
		this.type = type;
	}

}

File: Example/Factory-Code/RealEstate.cs
/// <summary>
/// @author Michael Hinterdorfer
/// </summary>
public abstract class RealEstate
{

    public RealEstate()
    {
    }

    private double price;

    private double squaremeter;

    private int rooms;

    private double garden_squaremeter;

    private int num_of_parkinglots;

    private Location location;


    public double getPrice()
    {
        return price;
    }

    public void setPrice(double price)
    {
        this.price = price;
    }

    public double getSquaremeter()
    {
        return squaremeter;
    }

    public void setSquaremeter(double squaremeter)
    {
        this.squaremeter = squaremeter;
    }

    public int getRooms()
    {
        return rooms;
    }

    public void setRooms(int rooms)
    {
        this.rooms = rooms;
    }

    public double getGarden_squaremeter()
    {
        return garden_squaremeter;
    }

    public void setGarden_squaremeter(double garden_squaremeter)
    {
        this.garden_squaremeter = garden_squaremeter;
    }

    public int getNum_of_parkinglots()
    {
        return num_of_parkinglots;
    }

    public void setNum_of_parkinglots(int num_of_parkinglots)
    {
        this.num_of_parkinglots = num_of_parkinglots;
    }

    public Location getLocation()
    {
        return location;
    }

    public void setLocation(Location location)
    {
        this.location = location;
    }

    public override string ToString()
    {
        string realestate = "";
        realestate += "Price: " + getPrice() + "\n";
        realestate += "Squaremeter: " + getSquaremeter() + "\n";
        realestate += "Garden Squaremeter: " + getGarden_squaremeter() + "\n";
        realestate += "Location: " + getLocation() + "\n";
        realestate += "Num of parkinglots: " + getNum_of_parkinglots() + "\n";
        realestate += "Rooms: " + getRooms() + "\n";
        realestate += "Type: " + GetType() + "\n";
        return realestate;
    }

}